From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Michael Goffioul <michael.goffioul@gmail.com>
Date: Wed, 23 Mar 2022 10:28:13 +0100
Subject: [PATCH 1/2] Fix virtgpu_virgl driver with non-ChromeOS kernel

Implementation of GRALLOC_DRM_GET_BUFFER_INFO for virtgpu_virgl relies
on custom DRM_IOCTL_VIRTGPU_RESOURCE_INFO_CROS ioctl handler in the
kernel, available in ChromeOS kernel. For non-ChromeOS kernel, simply
use the fallback implementation of resource_info in minigbm.

Change-Id: I8c01d1400ac287b3b35ca557f5d50f11b8fc7407
---
 virtgpu_virgl.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/virtgpu_virgl.c b/virtgpu_virgl.c
index 7c63b6c..ee79109 100644
--- a/virtgpu_virgl.c
+++ b/virtgpu_virgl.c
@@ -1056,6 +1056,7 @@ static void virgl_resolve_format_and_use_flags(struct driver *drv, uint32_t form
 	}
 }
 
+__attribute__((unused))
 static int virgl_resource_info(struct bo *bo, uint32_t strides[DRV_MAX_PLANES],
 			       uint32_t offsets[DRV_MAX_PLANES], uint64_t *format_modifier)
 {
@@ -1109,5 +1110,5 @@ const struct backend virtgpu_virgl = { .name = "virtgpu_virgl",
 				       .bo_flush = virgl_bo_flush,
 				       .resolve_format_and_use_flags =
 					   virgl_resolve_format_and_use_flags,
-				       .resource_info = virgl_resource_info,
+				       .resource_info = NULL,
 				       .get_max_texture_2d_size = virgl_get_max_texture_2d_size };
